src/django_components/slots.py

```diff
@@ -535,16 +533,17 @@ def on_node(entry: NodeTraverse) -> None:
         # - 0002: []
         # - 0003: [0004]
         # In other words, the data tells us that slot ID 0001 is PARENT of slot 0002.
-        curr_entry = entry.parent
-        while curr_entry and curr_entry.parent is not None:
```

Comment:
One part of the issue was in this `while` condition, which walks the nodes inside a component's template to find all `{% slot %}` tags.